core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/main/dot-edit-page-nav/directives/dot-edit-page-nav.directive.ts

```diff
@@ -54,7 +52,13 @@ export class DotEditPageNavDirective implements OnDestroy {
         this.destroy$.complete();
     }
 
-    private removePortletsClasess() {
+    private addPortletClass(url: string) {
+        const key = Object.keys(urlPortletRules).find((key) => url.includes(key));
```

Comment:
What if we made the `Object.keys(urlPortletRules)` a constant created on Init or in the constructor?